robotoff/app/api.py

```diff
@@ -233,38 +230,38 @@ def on_post(self, req: falcon.Request, resp: falcon.Response):
         self.spellcheck(req, resp)
 
     def spellcheck(self, req: falcon.Request, resp: falcon.Response):
-        text = req.get_param("text")
-        barcode = req.get_param("barcode")
+        text = self.__get_text(req, resp)
+        if text is None:
+            return
+
         index_name = req.get_param(
             "index", default=settings.ELASTICSEARCH_PRODUCT_INDEX
         )
         confidence = req.get_param_as_float("confidence", default=1.0)
+        spellchecker = Spellchecker(
+            client=es_client, index_name=index_name, confidence=confidence
+        )
 
-        if text is None and barcode is None:
-            raise falcon.HTTPBadRequest("text or barcode is required.")
-
-        if not text:
-            product = get_product(barcode) or {}
-            text = product.get("ingredients_text_fr")
+        resp.media = {}
```

Comment:
I find it easier to read if you assign resp.media one time:
```
resp.media = {
    "status": "success",
    "text": text,
    ...
}
```

Reply:
https://github.com/openfoodfacts/robotoff/pull/181/commits/13524631aa55dfcb63929ab779a0f761b947b51c